### PR TITLE
Initial local fetcher implementation.

### PR DESCRIPTION
**Pull request by saeta**
Implement the initial version of a local fetcher which an engine can
use to execute requests against local naptime resources.

### CODE REVIEW

**Comment by saeta** [comment]
PTAL @bryan-coursera

File: naptime-testing/src/test/scala/org/coursera/naptime/NestedMacroCourierTests.scala [hunk 1]
```diff
@@ -104,4 +118,100 @@ class NestedMacroCourierTests extends AssertionsForJUnit {
...
+    val naptimeRoutes = NaptimeRoutes(injector, routerBuilders)
+    val fetcher = new LocalFetcher(naptimeRoutes)
+
+    val request = Request(
+      requestHeader = FakeRequest(),
+      topLevelRequests = List(
+        TopLevelRequest(
+          CoursesResource.ID,
+          RequestField(
+            name = "get",
```
Comment (bryan-coursera) [inline naptime-testing/src/test/scala/org/coursera/naptime/NestedMacroCourierTests.scala]: should this name be multiGet?

File: naptime-testing/src/test/scala/org/coursera/naptime/NestedMacroCourierTests.scala [hunk 2]
```diff
@@ -104,4 +118,100 @@ class NestedMacroCourierTests extends AssertionsForJUnit {
...
```
Comment (bryan-coursera) [inline naptime-testing/src/test/scala/org/coursera/naptime/NestedMacroCourierTests.scala]: and this should be "search"?

File: naptime/src/main/scala/org/coursera/naptime/ari/fetcher/LocalFetcher.scala [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+/**
+ * Executes data requests against local Naptime resources (requires the use of Engine2 engines).
+ *
+ * @param naptimeRoutes The routing data structures required for handling requests.
+ */
+class LocalFetcher @Inject() (
+    naptimeRoutes: NaptimeRoutes)
+  extends FetcherApi
+    with StrictLogging {
```
Comment (bryan-coursera) [inline naptime/src/main/scala/org/coursera/naptime/ari/fetcher/LocalFetcher.scala]: unindent this by 2 i think

File: naptime/src/main/scala/org/coursera/naptime/ari/fetcher/LocalFetcher.scala [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        resourceSchema.name == topLevelRequest.resource.topLevelName &&
+          resourceSchema.version.contains(topLevelRequest.resource.version)
+      }
+      (for {
+        resourceSchema <- resourceSchemaOpt
+        router <- routers.get(resourceSchema.className)
+        argMap = topLevelRequest.selection.args.toMap
+        queryString = argMap.map { arg =>
+          var strRepr = arg._2.toString()
+          if (strRepr.charAt(0) == '"') {
```
Comment (bryan-coursera) [inline naptime/src/main/scala/org/coursera/naptime/ari/fetcher/LocalFetcher.scala]: probably a little safer to convert these three lines into:
```scala
strRepr.stripPrefix("\"").stripSuffix("\"")
```

File: naptime/src/main/scala/org/coursera/naptime/ari/fetcher/LocalFetcher.scala [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    }
+  }
+
+  private[this] def constructPath(resourceSchema: Resource, argMap: Map[String, JsValue]): String = {
+    val versionString = resourceSchema.version.map(v => s".v$v").getOrElse("")
+    val basePath = s"${resourceSchema.name}$versionString"
+
+    if (argMap.contains("id")) {
+      var idStr = Json.stringify(argMap("id"))
+      if (idStr.startsWith("\"")) {
```
Comment (bryan-coursera) [inline naptime/src/main/scala/org/coursera/naptime/ari/fetcher/LocalFetcher.scala]: same here about stripPrefix / stripSuffix

**Comment by bryan-coursera** [comment]
some small nits (and your tests failed), but once you get this passing and clean up the smaller things, LGTM

**Pull request merged**